### PR TITLE
Logical file

### PR DESCRIPTION
**Pull request by GordonSmith**
## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by GordonSmith** [comment]
Is dependent on:  https://github.com/hpcc-systems/Visualization/pull/3254

File: packages/comms/src/ecl/logicalFile.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    get binInfo(): string { return this.get("binInfo"); }
+    get PackageID(): string { return this.get("PackageID"); }
+    get Partition(): WsDfu.Partition { return this.get("Partition"); }
+    get Blooms(): WsDfu.Blooms { return this.get("Blooms"); }
+    get ExpireDays(): number { return this.get("ExpireDays"); }
+    get KeyType(): string { return this.get("KeyType"); }
+
+    get properties(): FileDetailEx { return this.get(); }
+
+    static attach(optsConnection: IOptions | IConnection | DFUService, Cluster: string, Name: string): LogicalFile {
```
Comment (jbrundage) [inline packages/comms/src/ecl/logicalFile.ts]: What is this attach method doing? Attaching one logical file to another?

Comment (GordonSmith) [reply packages/comms/src/ecl/logicalFile.ts]: Note that method is "static singleton factory" and that the constructor is "private" - so the pattern is:
```
const logicalFileRef = LogicalFile.attach(baseUrl, ID);
```

And the following is impossible:
```
const logicalFileRef = new LogicalFile(baseUrl, ID);
```

The idea is to ensure that there can only ever be 1 instance of the LogicalFile class per unique logical file.

Comment (jbrundage) [reply packages/comms/src/ecl/logicalFile.ts]: And then you'd use it like this?
```javascript
const baseUrl = "https://play.hpccsystems.com:18010";
const ID = ".::kaggle_county_crimes.csv";
const logicalFileRef = LogicalFile.attach(baseUrl, ID);
const infoObj = logicalFileRef.fetchInfo();
console.log(infoObj);
```

Comment (GordonSmith) [reply packages/comms/src/ecl/logicalFile.ts]: Nearly:
```
const logicalFileRef = LogicalFile.attach({baseUrl:  "https://play.hpccsystems.com:18010"}, ".::kaggle_county_crimes.csv");
...
```

**Pull request merged**